File: samples/java/TicTacToe/Server/src/main/java/systems/zlink/samples/tictactoe/server/configuration/SampleLocationStore.java
package systems.zlink.samples.tictactoe.server.configuration;

import systems.zlink.framework.locations.redis.ZLinkRedisLocationOptions;
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore;

import java.time.Duration;

public final class SampleLocationStore {
    private SampleLocationStore() {}

    public static ZLinkRedisLocationStore create(PlaySettings settings) {
        return create(settings.redisEndpoint(), settings.redisKeyPrefix());
    }

    public static ZLinkRedisLocationStore create(ApiSettings settings) {
        return create(settings.redisEndpoint(), settings.redisKeyPrefix());
    }

    private static ZLinkRedisLocationStore create(String redisEndpoint, String redisKeyPrefix) {
        return new ZLinkRedisLocationStore(
                new ZLinkRedisLocationOptions()
                        .setConnectionString(redisEndpoint)
                        .setKeyPrefix(redisKeyPrefix + "locations:")
                        .setCommandTimeout(Duration.ofMillis(500)));
    }
}

File: samples/kotlin/TicTacToe/Server/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/server/configuration/SampleLocationStore.kt
package systems.zlink.samples.kotlin.tictactoe.server.configuration

import java.time.Duration
import systems.zlink.framework.locations.redis.ZLinkRedisLocationOptions
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore

object SampleLocationStore {
    fun create(settings: SampleSettings): ZLinkRedisLocationStore =
        ZLinkRedisLocationStore(
            ZLinkRedisLocationOptions()
                .setConnectionString(settings.redisEndpoint)
                .setKeyPrefix(settings.redisKeyPrefix + "locations:")
                .setCommandTimeout(Duration.ofMillis(500))
        )
}
